src/grimp/adaptors/importscanner.py

```diff
@@ -10,20 +13,109 @@
 logger = logging.getLogger(__name__)
 
 
-class NotAnImport(Exception):
-    pass
+class _TreeWalker(NodeVisitor):
+    def __init__(self,
+                 import_scanner: AbstractImportScanner,
+                 module: Module,
+                 found_package: FoundPackage,
+                 module_lines: List[str],
+                 is_package: bool) -> None:
+        self.module = module
+        self.found_package = found_package
+        self._import_scanner = import_scanner
+        self.module_lines = module_lines
+        self.is_package = is_package
+        self.direct_imports: Set[DirectImport] = set()
+
+        self.is_type_checking = False
+        super().__init__()
+
+    @property
+    def found_packages(self) -> Set[FoundPackage]:
+        return self._import_scanner.found_packages
+
+    @property
+    def include_external_packages(self) -> bool:
+        return self._import_scanner.include_external_packages
+
+    def visit_Import(self, node: Import) -> None:
+        self._parse_direct_imports_from_node(node, _ImportNodeParser)
+
+    def visit_ImportFrom(self, node: ImportFrom) -> None:
+        self._parse_direct_imports_from_node(node, _ImportFromNodeParser)
+
+    def visit_If(self, node: If) -> None:
+        cond = False
+        if isinstance(node.test, Attribute):
+            # Case for "if t.TYPE_CHECKING:" or "if typing.TYPE_CHECKING:"
+            cond = node.test.attr == "TYPE_CHECKING"
```

Comment:
You don't test the left part: it should be `typing` or `t`

Reply:
Changed the comment. I explicitly want to test for any alias